File: taskapp-ser/src-cs/task/app/TaskService.cs
using System;
using System.Collections.Generic;
using byps;

namespace task.app
{
	
	// checkpoint byps.gen.cs.GenRemoteClass:93
	public interface TaskService : BRemote {
	
		void AddTask(TaskInfo task);
		void AddTask(TaskInfo task, BAsyncResult<Object> asyncResult) ;
		
		IList<TaskInfo> GetTasks();
		void GetTasks(BAsyncResult<IList<TaskInfo>> asyncResult) ;
		
		void RegisterCalculationService(CalculationService calc);
		void RegisterCalculationService(CalculationService calc, BAsyncResult<Object> asyncResult) ;
		
		CalculationService GetCalculationService();
		void GetCalculationService(BAsyncResult<CalculationService> asyncResult) ;
		
		IList<System.IO.Stream> GetTaskAttachments(int taskId);
		void GetTaskAttachments(int taskId, BAsyncResult<IList<System.IO.Stream>> asyncResult) ;
		
		
	}
}  // end namespace

File: taskapp-api/src/task/app/TaskService.java
package task.app;

import java.io.InputStream;
import java.util.List;

import byps.Remote;
import byps.RemoteException;

public interface TaskService extends Remote {

	void addTask(TaskInfo task) throws RemoteException;
	
	List<TaskInfo> getTasks() throws RemoteException;
	
	/**
	 * @since 1.0.0.1
	 */
	void registerCalculationService(CalculationService calc) throws RemoteException;
	
	/**
	 * @since 1.0.0.1
	 */
	CalculationService getCalculationService() throws RemoteException;
	
	/**
	 * @since 1.0.0.1
	 */
	List<InputStream> getTaskAttachments(int taskId) throws RemoteException;
}
